be/src/vec/sink/vtablet_finder.cpp

```diff
@@ -95,8 +95,22 @@ Status OlapTabletFinder::find_tablets(RuntimeState* state, Block* block, int row
     if (_find_tablet_mode == FindTabletMode::FIND_TABLET_EVERY_ROW) {
         _vpartition->find_tablets(block, qualified_rows, partitions, tablet_index);
     } else {
+        // for random distribution
         _vpartition->find_tablets(block, qualified_rows, partitions, tablet_index,
                                   &_partition_to_tablet_map);
+        if (_find_tablet_mode == FindTabletMode::FIND_TABLET_EVERY_BATCH) {
```

Comment:
`FIND_TABLET_EVERY_BATCH` can save time because it only find tablets once for each batch.
But after you change, it still iterate each row in batch to update the partition's `load_tablet_idx`,
may it impact the performance?

Reply:
the main contribution of random  is to send the whole data batch to one tablet, not save the find tablet time, which operation is light, and here  _partition_to_tablet_map size is small and not the bottleneck for distributing data